nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,2575 @@
+      },
+      "source": [
+        "print(\"Text: {} \\nLabel: {}\".format(synthetic_10_ds['text'][5], idx2label[synthetic_10_ds['label'][5]]))"
+      ],
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Text:  even if ur not into these kind of things u have to admit it's pretty cool  \n",
+            "Label: joy\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "TiofN18bKZ0f",
+        "outputId": "4f0d716c-49c2-48f7-8a27-cab0daf0077e"
+      },
+      "source": [
+        "print(\"Text: {} \\nLabel: {}\".format(synthetic_50_ds['text'][5], idx2label[synthetic_50_ds['label'][5]]))"
+      ],
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Text:  i want to stop running and walk...but the fact that i'm still running is the real miracle  \n",
+            "Label: joy\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "RpwH-elSKaKh",
+        "outputId": "577b2b0f-f04f-40ea-ffe1-4e268df03bf1"
+      },
+      "source": [
+        "print(\"Text: {} \\nLabel: {}\".format(synthetic_100_ds['text'][5], idx2label[synthetic_100_ds['label'][5]]))"
+      ],
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Text:  i want a beer right now  \n",
+            "Label: anger\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "M6dLc7xMK0NJ"
+      },
+      "source": [
+        "We see that GPT-3 has effectively generated very realistic samples. 👏👏👏"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gesWuhM9jbLq"
+      },
+      "source": [
+        "## 🚀 Model "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_QaSNzfSTG4D"
+      },
+      "source": [
+        "Here we define the model and the training pipeline. We will use [DistilBERT](https://arxiv.org/abs/1910.01108) that is a light Transformer trained by distilling BERT base. It has 40% less parameters than bert-base-uncased, runs 60% faster while preserving over 95% of BERT’s performances as measured on the GLUE language understanding benchmark."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "0xqcEiJqBZsB"
+      },
+      "source": [
+        "metric = load_metric(\"accuracy\")\n",
+        "\n",
+        "batch_size = 6\n",
+        "epochs = 20\n",
```

Comment:
epochs actually isn't the best factor for determining running length here 🤔 .
because if you think about it, the augmented dataset is then larger than the original, so finetuning on the augmented model will have more chances to do backprop. So actually, we want to fix the number of steps instead of the number of epochs. 
The number of steps is the steps needed to have at least one (preferably more) full passes of the largest augmented datasets.

So suppose you have dataset A of 100 datapoints
and dataset B of 500 datapoints

and a batch size of 10

Then you might want to do 5 full passes on the largest dataset
So the number of steps for dataset B would be: 5 * (500/10) = 250

With this number of steps, we would have 25 epochs on dataset A

Which is only fair